Add stripack to build_visit

markcmiller86 opened the pull request:
stripack is a bit unusuall for build_visit TPLs...

1. It is [ACM Licensed code](https://www.acm.org/publications/policies/software-copyright-notice).
1. It is used in VisIt as a dynamically loaded library only in ffp plugin. So, it isn't installed or managed like a typical TPL dependency.
1. It is a fortran library.

This update makes it possible (and easy) for users who want it and accept its license terms to use build_visit to build and install libstripack for the ffp plugin.

markcmiller86 committed:
remove DO_SVN block

markcmiller86 committed:
replace DO_SVN with DO_GIT

markcmiller86 committed:
add stripack

markcmiller86 committed:
fix license prompt behavior

markcmiller86 committed:
fix license prompt behavior

markcmiller86 commented:
I've asked @cyrush to take a quick look for licensing handling.

brugger1 commented on src/tools/dev/scripts/bv_support/bv_stripack.sh:
```diff
@@ -0,0 +1,171 @@
+    if [[ "$DO_STRIPACK" == "yes" ]] ; then
+        ensure_built_or_ready "stripack" $STRIPACK_VERSION $STRIPACK_BUILD_DIR $STRIPACK_FILE
+        if [[ $? != 0 ]] ; then
+            warn "Unable to build google perftools.  ${STRIPACK_FILE} not found."
```
This error message looks incorrect. It mentions google perftools. It should probably mention stripack.

brugger1 commented on src/tools/dev/scripts/bv_support/bv_tcmalloc.sh:
```diff
@@ -62,10 +62,10 @@ function bv_tcmalloc_ensure
             warn "Unable to build google perftools.  ${TCMALLOC_FILE} not found."
             ANY_ERRORS="yes"
             DO_TCMALLOC="no"
-            if [[ "$DO_SVN" != "yes" ]] ; then
```
I don't know what it means to download this through git (or svn as it was before). It seems like this could be removed altogether. I see that it isn't in our third-party repo. I think we should add it to the third-party directory. You can submit this as another ticket if too much effort.

markcmiller86 replied:
FYI...I think google-perftools was somewhat *special* in that its download was *restricted* to only those who had non-anonymous SVN repo read access. I don't know why that was the case but my belief is based on some of the comments I read here and elsewhere. So, I dunno if right thing is to add it to our TPL repo as that would certainly be open to anonymous download.

markcmiller86 committed:
minor fixes

markcmiller86 committed:
Replace OSX-specifc -dwx test syntax

markcmiller86 committed:
Merge branch '3.1RC' into task-mcm86-16Feb20-build-stripack

markcmiller86 committed:
adding logic to copy stripack to build/lib

markcmiller86 committed:
Merge branch 'task-mcm86-16Feb20-build-stripack' of github.com:visit-dav/visit into task-mcm86-16Feb20-build-stripack

markcmiller86 committed:
rebase due to orig. gen on OSX

markcmiller86 commented:
I've added two items to this PR to fix testing.

1. Added a line of logic to copy `libstripack.so` to visit build point's `lib` dir so ffp plugin will see it and gen `ffp_03.png` test result.
1. Re-based `ffp_02.png` test result.

After merging this PR, `ffp_03.png` may still fail and if it does, it will likey be because I did original gen on OSX. So, it should just need to be rebased if so.

I will be in training all day today and will try to merge this after.

@cyrush if you have a moment to render opinion w.r.t. licensing of stripack here and how I've handled it, that would be great.

markcmiller86 commented:
I've added two items to this PR to fix testing.

1. Added a line of logic to copy `libstripack.so` to visit build point's `lib` dir so ffp plugin will see it and gen `ffp_03.png` test result.
1. Re-based `ffp_02.png` test result.

After merging this PR, `ffp_03.png` may still fail and if it does, it will likey be because I did original gen on OSX. So, it should just need to be rebased if so.

I will be in training all day today and will try to merge this after.

@cyrush if you have a moment to render opinion w.r.t. licensing of stripack here and how I've handled it, that would be great.

markcmiller86 commented:
I've added two items to this PR to fix testing.

1. Added a line of logic to copy `libstripack.so` to visit build point's `lib` dir so ffp plugin will see it and gen `ffp_03.png` test result.
1. Re-based `ffp_02.png` test result.

After merging this PR, `ffp_03.png` may still fail and if it does, it will likey be because I did original gen on OSX. So, it should just need to be rebased if so.

I will be in training all day today and will try to merge this after.

@cyrush if you have a moment to render opinion w.r.t. licensing of stripack here and how I've handled it, that would be great.

markcmiller86 commented:
I've added two items to this PR to fix testing.

1. Added a line of logic to copy `libstripack.so` to visit build point's `lib` dir so ffp plugin will see it and gen `ffp_03.png` test result.
1. Re-based `ffp_02.png` test result.

After merging this PR, `ffp_03.png` may still fail and if it does, it will likey be because I did original gen on OSX. So, it should just need to be rebased if so.

I will be in training all day today and will try to merge this after.

@cyrush if you have a moment to render opinion w.r.t. licensing of stripack here and how I've handled it, that would be great.

markcmiller86 committed:
Merge branch '3.1RC' into task-mcm86-16Feb20-build-stripack

cyrush commented:
@markcmiller86 this looks sensible to me. 

Folks will have to agree to the license when they build. 
Our use for testing and is non-commercial.

markcmiller86 commented:
Thanks @cyrush for the sanity check.
